lib/Tooling/CrossTranslationUnit.cpp

```diff
@@ -262,6 +262,9 @@ const FunctionDecl *CrossTranslationUnit::getCrossTUDefinition(
           findFunctionInDeclContext(TU, LookupFnName)) {
     auto *ToDecl = cast_or_null<FunctionDecl>(
         Importer.Import(const_cast<FunctionDecl *>(ResultDecl)));
+    if (Context.getDiagnostics().hasErrorOccurred()) {
```

Comment:
Could you give a short example in a comment why this change is necessary? I remember that it was needed for some reason a long time ago, but using my fork the checking works without this change. Probably the projects I check don't run into this error.

Reply:
We used this change to be able to reproduce some ASTImporter errors with creduce.
The problem was that creduce created such .cpp files which were started to have ODR errors when we merged them together. And this way the reduction went in a wrong direction because such output was created which was invalid.
Actually we should check also the warnings, because now ODR violation is diagnosed as a waning.

Reply:
And does this `hasErrorOccurred()` function catch the warnings too? So is this still a relevant pull request?

Reply:
No. We should use `getNumWarnings()` too.